cordova-lib/src/cordova/project_metadata.js

```diff
@@ -0,0 +1,60 @@
+/**
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+var cordova_util = require('./util'),
+    ConfigParser = require('../configparser/ConfigParser'),
+    Q            = require('q');
+
+// Returns all the platforms that are currently saved into config.xml
+// Return type is a promise that is fulfilled with a list of objects with name and version properties. e.g: [{name: 'android', version: '3.5.0'}, {name: 'wp8', version: 'C:/path/to/platform'}, {name: 'ios', version: 'git://...'}]
+// ToDo: Once we move to npm, this function should be updated to rely on npm instead
+function getPlatforms(projectRoot){
+    var xml = cordova_util.projectConfig(projectRoot);
+    var cfg = new ConfigParser(xml);
+    return Q(cfg.getEngines());
+}
+
+// Returns all the plugins that are currently saved into config.xml
+// Return type is a promise that is fulfilled with a list of objects with name and version properties. e.g: [ {id: 'org.apache.cordova.device', name: 'Device', APP_ID: 'my-app-id', APP_NAME: 'my-app-name'} ]
+// ToDO: Once we move to npm, this function should be updated to rely on npm instead
+function getPlugins(projectRoot){
```

Comment:
Should we return promises from an API? This couples our implementation with Q. We should ideally return the data itself and document the shape of the data rather than return a Q based promise